gateway-server/src/main/java/org/apache/knox/gateway/webshell/ProcessConnectInfo.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.knox.gateway.webshell;
+
+import org.eclipse.jetty.io.RuntimeIOException;
+import de.thetaphi.forbiddenapis.SuppressForbidden;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+/**
+* data structure to store a connection session
+*/
+public class ProcessConnectInfo extends ConnectInfo {
+
+
+    private Process process;
+    private static final Logger LOG = LoggerFactory.getLogger(ProcessConnectInfo.class);
+
+    @SuppressForbidden
+    @Override
+    public void connect(String username){
+        try {
+            ProcessBuilder builder = new ProcessBuilder("sudo","-u",username,"-H","bash","-i");
+            builder.redirectErrorStream(true); // combine stderr with stdout
+            process = builder.start();
+            inputStream = process.getInputStream();
+            outputStream = process.getOutputStream();
```

Comment:
I feel we shold actively save references to all the processes created in events like 
1. We need to terminate them safely
2. Make sure we don't spwan too many of them (throttle may be)
3. For future use display them
4. 4. Control them from api/ui